account_avatax_oca/models/account_move.py

```diff
@@ -176,7 +176,12 @@ def onchange_reset_avatax_amount(self):
     # Same as v12
     def get_origin_tax_date(self):
         if self.invoice_doc_no:
-            orig_invoice = self.search([("name", "=", self.invoice_doc_no)])
+            orig_invoice = self.search(
+                [
+                    ("name", "=", self.invoice_doc_no),
+                    ("partner_id", "=", self.partner_id.id),
```

Comment:
Would this be better done by Commercial Partner ID?
(I'm aware Odoo OOB is not great at this, but still, let's not contribute to the problem...)